warrior/WarriorCore/Classes/kw_driver_class.py

```diff
@@ -157,33 +158,66 @@ def get_mandatory_arguments(self):
             args.remove('self')
         return args
 
+    def get_credential_value(self, arg, system):
+        """get the value of arg in data file corresponding to system
+        """
+        datafile = self.data_repository['wt_datafile']
+        var = arg
+        if not hasattr(self, 'tag_dict'):
+            self.tag_dict = data_Utils.get_credentials(datafile, system)
+            arg = arg.strip()
+        if isinstance(arg, basestring) and arg.startswith("wtag"):
+            var = arg.split("=")[1].strip()
+        if var in self.tag_dict:
+            value = self.tag_dict[var]
+            # substitute environment/datarepo variables in the value and return
+            if isinstance(value, (basestring, list, dict)):
+                return data_Utils.substitute_var_patterns(value)
+            else:
+                return value
+        else:
+            return None
```

Comment:
If var is not found in tag_dict, it should print an error message
please add one here unless there is an error message print somewhere else associate with this

Reply:
It is handled in get_values_for_(mandatory|optional)_args...